docker/build-tools/build-and-push.sh

```diff
@@ -49,6 +49,18 @@ ARCH="${TARGET_ARCH:-amd64}"
 # Generally, this should always be set even with a single architecture build or we will end up with only an image with a `-{arch}` suffix.
 MANIFEST_ARCH="${MANIFEST_ARCH-amd64}"
 
+# CACHE_FROM_TAG, if present, defines an image tag which is used in the build --cache-from option.
+# Allow overriding for other tools like podman
+CACHE_FROM_TAG="${CACHE_FROM_TAG:-:master-latest-amd64}"
+
+# The podman build --cache-from option value must contain neither a tag nor digest.
+# Allow overriding of the CACHE_FROM_TAG by setting an empty string.
+if [[ "${CONTAINER_CLI}" == "podman" ]]; then
+  CACHE_FROM_TAG=""
+else
+  CACHE_FROM_TAG=":${BRANCH}-latest-${ARCH}"
+fi
```

Comment:
I'd do the opposite: Initialize it unconditionally with `":${BRANCH}-latest-${ARCH}"` (before the `if`), and then do the check: if it's podman then clear it.

Reply:
Yes, it's a different behavior between podman and docker using the build `--cache-from` flag. I think the problem is in :
https://github.com/containers/podman/blob/main/cmd/podman/images/build.go#L565
https://github.com/containers/buildah/blob/main/pkg/parse/parse.go#L64